auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,108 @@
+            AuditlogRule, self.with_context(real_user=self.env.context.get('uid'))
+            ).sudo(user=user)
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.multi
+    def unlink(self):
+        lines = self.mapped("auditlog_line_access_rule_ids")
+        res = super(AuditlogRule, self).unlink()
+        if res:
+            lines.unlink()
+        return res
+
+    @api.multi
+    def _get_view_log_lines_action(self):
+        self.ensure_one()
+        fields_ids = self.auditlog_line_access_rule_ids.mapped('field_ids').ids
+        logs = self.env['auditlog.log'].sudo().search([
+            ('model_id', '=', self.model_id.id),
+            ('res_id', 'in', self.env.context.get('active_ids'))
+        ])
+        print('XXX', logs.ids)
+        lines = self.env['auditlog.log.line'].sudo().search([
+            ('log_id', 'in', logs.ids), ('field_id', 'in', fields_ids)
+        ])
+        print('YYY', lines.ids)
+        return {
+            "name": _("View Log Lines"),
+            "res_model": "auditlog.log.line",
+            #"src_model": self.model_id.model,
+            #"binding_model_id": self.model_id.id,
+            "view_mode": "tree,form",
+            "view_id": False,
+            "domain": [('id', 'in', lines.ids)],
+            "type": "ir.actions.act_window",
+        }
+
+    @api.multi
+    def _create_server_action(self):
+        self.ensure_one()
+        code = \
+            "rule = env['auditlog.rule'].browse(%s)\n" \
+            "action = rule._get_view_log_lines_action()" % (self.id,)
+        server_action = self.env['ir.actions.server'].sudo().create({
+            'name': "View Log Lines",
+            'model_id': self.model_id.id,
+            'state': "code",
+            'code': code
+        })
+        self.write({
+            'server_action_id': server_action.id
+        })
+        return server_action
+
+    @api.multi
+    def subscribe(self):
+        for rule in self:
+            server_action = rule._create_server_action()
+            server_action.create_action()
+        return super(AuditlogRule, self).subscribe()
+
+    @api.multi
+    def unsubscribe(self):
+        for rule in self:
+            rule.server_action_id.unlink()
+        return super(AuditlogRule, self).unsubscribe()
+
+    def _prepare_log_line_vals_on_read(self, log, field, read_values):
+        res = super(AuditlogRule, self)._prepare_log_line_vals_on_read(
+            log, field, read_values)
+        res.update({'user_id': self.env.context.get('real_user')})
+        return res
+
+    def _prepare_log_line_vals_on_write(
+            self, log, field, old_values, new_values):
+        res = super(AuditlogRule, self)._prepare_log_line_vals_on_write(
+            log, field, old_values, new_values)
+        res.update({'user_id': self.env.context.get('real_user')})
```

Comment:
Actually instead of doing this here, can you debug why this:

https://github.com/OCA/server-tools/blob/11.0/auditlog/models/rule.py#L375

is not resulting in the correct value, and then change it?

I mean it's doing the `sudo` here, but before that, it's storing the active `uid`, so that should already be working.

https://github.com/OCA/server-tools/blob/11.0/auditlog/models/rule.py#L231

Checked on test server and it is:

```
odoo-inuka-testdatabase=# select user_id, count(1) from auditlog_log where create_date >= '2022-07-01' group by user_id;
 user_id | count 
---------+-------
       1 |   181
     239 |  1862
  163050 |    38
(3 rows)

odoo-inuka-testdatabase=# select id, login from res_users where id in (1, 239, 163050);
   id   |       login        
--------+--------------------
      1 | system...
    239 | J....
 163050 | T....
(3 rows)
```

Reply:
It is not unfortunately. the methods in auditlog patch the core write, create methods on module initialization , rendering unaccessible the real user when overwriting auditlog.log.line crud methods. the same for all other auditlog.log.line methods.

In the auditlog.log code, instead the "real" user_id is available will solve by referencing that.

Reply:
7eff32c5250b9df0e2edf1e10b3ae2313c711873     also allows to remove a bit of code.

Reply:
Great. Note: maybe we need to make that one stored, required and indexed as well, and include the migration script for it, in case it's used in a searchquery somewhere. (Let's not do that now, but i leave the note here just in case).